libs/build-sqlcipher-ios.sh

```diff
@@ -59,33 +59,16 @@ CFLAGS="\
   -mios-version-min=${IOS_MIN_SDK_VERSION} \
 "
 
-# Keep in sync with SQLCIPHER_CFLAGS in `build-sqlcipher-desktop.sh` for now (it probably makes
-# sense to try to avoid this duplication in the future).
-# TODO: We could probably prune some of these, and it would be nice to allow debug builds (which
-# should set `SQLITE_DEBUG` and `SQLITE_ENABLE_API_ARMOR` and not `NDEBUG`).
+# Match the SQLCIPHER_CFLAGS used in Firefox for iOS v15.x and earlier.
 SQLCIPHER_CFLAGS=" \
-  -DSQLITE_HAS_CODEC \
-  -DSQLITE_SOUNDEX \
-  -DHAVE_USLEEP=1 \
-  -DSQLITE_MAX_VARIABLE_NUMBER=99999 \
-  -DSQLITE_THREADSAFE=1 \
-  -DSQLITE_DEFAULT_JOURNAL_SIZE_LIMIT=1048576 \
   -DNDEBUG=1 \
```

Comment:
Nit: keep
- `-DNDEBUG` (which isn't considered a sqlite compile option)
- `-DSQLITE_ENABLE_JSON1` (we've talked about using this in the past, and it would be an annoying gotcha to forget it's not present in the iOS version),
- `-DSQLITE_MAX_VARIABLE_NUMBER=99999` (the default of 999 makes us do some dumb things in the places db)
- `-DSQLITE_SECURE_DELETE` (lockbox will want this).

I also think we should consider keeping:
- `-DSQLITE_DEFAULT_JOURNAL_SIZE_LIMIT=1048576`
- `-DSQLITE_DEFAULT_PAGE_SIZE=32768` `-DSQLITE_MAX_DEFAULT_PAGE_SIZE=32768`